src/inotify.cr

```diff
@@ -1,31 +1,47 @@
 require "./inotify/version"
 require "./inotify/event"
-require "./inotify/fallback"
+
+{% skip_file unless flag?(:linux) %}
+
+require "./inotify/lib_inotify"
 require "./inotify/settings"
+require "./inotify/watcher"
 
-{% if flag?(:linux) %}
-  require "./inotify/lib_inotify"
-  require "./inotify/watcher"
-{% else %}
-  module Inotify
-    alias Watcher = Fallback
+module Inotify
+  def self.watcher(recursive : Bool = false) : Inotify::Watcher
+    Watcher.new(recursive)
   end
-{% end %}
 
-module Inotify
-  def self.watch(path : String, recursive : Bool = false, &block : Event ->)
-    Watcher.new(path, recursive, &block)
+  def self.watch(path : String, recursive : Bool = false, &block : Inotify::Event ->) : Inotify::Watcher
+    inotify = Inotify.watcher recursive
+    inotify.on_event &block
+    inotify.watch path
+    inotify
   end
 end
 
 class File
```

Comment:
This monkey patching can be easily avoided, this are class method.
For `#watch`, a method can accept a path or `File`.

Reply:
I remove them, they just pollute std-lib `File` and `Dir`.